src/builder.rs

```diff
@@ -0,0 +1,831 @@
+        self.kind = kind.into();
+        self
+    }
+
+    pub fn name<VALUE: Into<String>>(&mut self, name: VALUE) -> &mut Self {
+        self.name = Some(name.into());
+        self
+    }
+
+    pub fn name_opt<VALUE: Into<Option<String>>>(&mut self, name: VALUE) -> &mut Self {
+        self.name = name.into();
+        self
+    }
+
+    pub fn uid<VALUE: Into<String>>(&mut self, uid: VALUE) -> &mut Self {
+        self.uid = Some(uid.into());
+        self
+    }
+
+    pub fn uid_opt<VALUE: Into<Option<String>>>(&mut self, uid: VALUE) -> &mut Self {
+        self.uid = uid.into();
+        self
+    }
+
+    /// Can be used to initialize a builder with settings from an existing resource.
+    /// The builder will create an `OwnerReference` that points to the passed resource.
+    ///
+    /// This will _not_ set `controller` or `block_owner_deletion`.
+    pub fn initialize_from_resource<T: Resource<DynamicType = ()>>(
+        &mut self,
+        resource: &T,
+    ) -> &mut Self {
+        self.api_version(T::api_version(&()))
+            .kind(T::kind(&()))
+            .name(resource.name())
+            .uid_opt(resource.meta().uid.clone());
+        self
+    }
+
+    pub fn build(&self) -> OperatorResult<OwnerReference> {
+        Ok(OwnerReference {
+            api_version: match self.api_version {
+                None => return Err(Error::MissingObjectKey { key: "api_version" }),
+                Some(ref api_version) => api_version.clone(),
+            },
+            block_owner_deletion: self.block_owner_deletion,
+            controller: self.controller,
+            kind: match self.kind {
+                None => return Err(Error::MissingObjectKey { key: "kind" }),
+                Some(ref kind) => kind.clone(),
+            },
+            name: match self.name {
+                None => return Err(Error::MissingObjectKey { key: "name" }),
+                Some(ref name) => name.clone(),
+            },
+            uid: match self.uid {
+                None => return Err(Error::MissingObjectKey { key: "uid" }),
+                Some(ref uid) => uid.clone(),
+            },
+        })
+    }
+}
+
+/// A builder to build [`Pod`] objects.
+///
+#[derive(Clone, Default)]
+pub struct PodBuilder {
+    metadata: Option<ObjectMeta>,
+    node_name: Option<String>,
+    tolerations: Vec<Toleration>,
+    status: Option<PodStatus>,
+    containers: Vec<Container>,
+}
+
+impl PodBuilder {
+    pub fn new() -> PodBuilder {
+        PodBuilder::default()
+    }
+
+    pub fn metadata_default(&mut self) -> &mut Self {
+        self.metadata(ObjectMeta::default());
+        self
+    }
+
+    pub fn metadata_builder<F>(&mut self, f: F) -> OperatorResult<&mut Self>
+    where
+        F: Fn(&mut ObjectMetaBuilder) -> &mut ObjectMetaBuilder,
+    {
+        let mut builder = ObjectMetaBuilder::new();
+        let builder = f(&mut builder);
+        self.metadata = Some(builder.build()?);
```

Comment:
Here you could just reuse the inner `ObjectMetaBuilder`  without calling `build()` on it and possibly overwrite previous metadata.